### ISSUE TITLE
Unifying Diffusion Models' Latent Space, with Applications to CycleDiffusion and Guidance

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#188]
# TL; DR

- Introduce a new formulation that frames diffusion models as deterministic mapping from a Gaussian prior.
- By forming a data-independent prior distribution, stochastic diffusion models can perform generic unpaired image translation and guided sampling.

# Motivation

- There has been little study on the latent space of the diffusion model, and it was commonly thought to be data-dependent (or random seed-dependent).
- This view of diffusion latent space significantly constrained applications natural in other generative models such as GANs and VAEs.

# Formulation

### Unifying the latent space

![image](https://user-images.githubusercontent.com/46308407/201339617-9307ebff-e4d9-4486-a7ff-8a234a216c4c.png)

- Unify various denoising generative models into a single group of deterministic map from a generic Gaussian prior.

![image](https://user-images.githubusercontent.com/46308407/201339769-ea7f5eb9-c018-4919-8420-891d60dad806.png)

- Formulate Gaussian priors for stochastic DPMs by extending the latent code (concatenating X_T and all the errors induced from the denoising steps, this is roughly equivalent to inferring the random seed from an image)

### DPM Encoder

![image](https://user-images.githubusercontent.com/46308407/201340276-e0d84ddf-1795-45de-8ce6-c423502fd49f.png)

- This DPM encoder relies on the diffusion model, as the model has to predict the denoised version to compute the error. These errors make up to find the random seed.

### Cycle Diffusion

- At this point, unpaired image translation is trivial because one can simply encode an image (xT + e1 + ... + eN) with the first diffuser and decode with a different one. 
- This is similar in spirit to DDIB (Dual Diffusion Implicit Bridges), but this only worked for deterministic diffusion (DDIM), because inverting an image with a stochastic (Markovian) diffusion process was non-trivial.
- Using a text-to-image model, we can perform text-based image editing by giving separate text prompts in the encoding and the decoding process.

### Plug-and-play Guidance

- When the diffusion model's latent space was believed to be data-dependent (or Markovian process-dependent), the guidance model (ex: CLIP) had to be trained to be explicitly aware of the noise schedule.
- This means the classifier had to be trained on noised images (with noise step as conditional input).
- However, now that diffusion model's (extended) latent space is simply Gaussian just like any other generative models (GANs), a generic guidance model can lead conditional diffusion sampling.
- Perform Langevin sampling in the extended Gaussian latent space, find the code that corresponds to the desired condition, then run diffusion process.

![image](https://user-images.githubusercontent.com/46308407/201341161-1e80d421-7a9d-4bd9-98cd-b2c855c9121e.png)

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/201341194-91bfd9b8-320a-464b-953b-28aa556f740a.png)
![image](https://user-images.githubusercontent.com/46308407/201341206-4b604436-4582-45e3-8e73-f51e98a99218.png)
![image](https://user-images.githubusercontent.com/46308407/201341226-017ca40e-4486-4ff1-a5b3-e0ff0685945e.png)
![image](https://user-images.githubusercontent.com/46308407/201341256-bf911194-5820-4e79-8a58-ba78a5141ebd.png)
![image](https://user-images.githubusercontent.com/46308407/201341388-82613ca9-73b1-4ae1-bc3f-9ccdd09b11ab.png)
![image](https://user-images.githubusercontent.com/46308407/201341423-3d1b034f-13c3-44bf-a8fc-89c84852ebd1.png)
![image](https://user-images.githubusercontent.com/46308407/201341430-f3abd569-f4fe-4d88-9489-c9cdb4a9fdee.png)

# Inspiration

- This paper is honestly somewhat messy at the first sight, but provides valuable insights into the latent space of diffusion models.
- Though time consuming, if we can extend the latent space to include all the errors made by the model in denoising steps, this is certainly Gaussian and puts a stochastic diffuser as a deterministic mapping (like GANs).
- No need to fear or fantasize about the latent space of stochastic diffusion models any more. They are just like GANs with even better properties (optimal transport?)
- Text-based editing was so cool.
